k8s/provider.go

```diff
@@ -137,3 +146,30 @@ func (p *ServiceProvider) client() (Client, error) {
 		k8sClient: client,
 	}, nil
 }
+
+func getIPRetry(ctx context.Context, client Client, podNamespace, podName string) (host string, err error) {
+	ch := make(chan string, 1)
+	go func() {
+		for {
+			pod, err := client.GetPod(ctx, podNamespace, podName)
```

Comment:
This is the second call to `client.GetPod`, one of them is redundant. IMO this function shoud be renamed to `getPodWithRetry() (*v1.pod, error)`, and return the verified pod. Any call to GetPod before this function verifies it, should be considered faulty. Any call to GetPod should be replaced with this.

Reply:
That sounds neat.